### DISCUSSION TITLE
Fabric API get flagged as a Rat

### DISCUSSION THREAD

**Discussion by phyrobyte** [FabricMC/fabric-api#3266] [FAQ] [answered]
Hi there, I would like to ask if Fabric Api is safe as it gets flagged as a rat on  isthisarat.com. I would like to ask because I don't know whether the bleeding pipe exploit or any other exploit has affected it. This may seem stupid for me to ask but I've been ratted once already and want to be extra sure. Even more stupid, I'm using 1.17.1 fabric api to make a mod and 1.17.1 was before all of these security, tysm

**Comment by modmuss50** [comment, marked as answer]
Its not as long as you make sure you download it from one from the official download locations (CurseForge, Modrinth or Github). 

I would not trust that "isthisarat.com" website at all. There is no infomation at all on that website about how it works, who made it or any indication that it would actually work.

Websites such Virus Total are a better place to check as they aggragate a wide range of industry standard malware detction suites, but are still far from bullet proof.

If you are still unsure you can build mods from source.

**Comment by phyrobyte** [comment]
Thanks for the help, I’ve been ratted before and just need make sure it doesn’t happen again
